tests/analysis/test_holes_characterization.py

```diff
@@ -60,3 +63,180 @@ def test_little_mcar_test(df_input: pd.DataFrame, expected: bool, request):
+def test__generate_label_feature_combinations(request, matrix_fixture):
+    X = request.getfixturevalue(matrix_fixture)
+    _, n_cols = X.shape
+    mcar_test_pklm = PKLMTest()
+    result = mcar_test_pklm._generate_label_feature_combinations(X)
+    # Check that number of projections is smaller than possible
+    assert len(result) <= mcar_test_pklm._get_max_draw(n_cols)
+    # Check there are no duplicates
+    assert len(set([x for x in result if result.count(x) > 1])) == 0
+    for features, label in result:
+        assert isinstance(label, int)
+        assert isinstance(features, list)
+        assert label not in features
+        for feature_index in features:
+            assert 0 <= feature_index <= (n_cols-1)
+
+
+
+@pytest.mark.parametrize("dataframe_fixture, features_idx, target_idx",
+    [
+        ("np_matrix_with_nan_mcar", np.array([1, 0]), 2),
+    ]
+)
+def test__build_dataset(request, dataframe_fixture, features_idx, target_idx):
+    dataframe = request.getfixturevalue(dataframe_fixture)
+    mcar_test_pklm = PKLMTest()
+    X, y = mcar_test_pklm._build_dataset(dataframe, features_idx, target_idx)
+    assert X.shape[0] == len(y)
+    assert not np.any(np.isnan(X))
+    assert not np.any(np.isnan(y))
+    assert np.all(np.unique(y) == [0, 1])
+    assert X.shape[1] == len(features_idx)
+    assert len(y.shape) == 1
+
+
+@pytest.mark.parametrize("dataframe_fixture, permutation_fixture, features_idx, target_idx",
+    [
+        ("np_matrix_with_nan_mcar", "missingness_matrix_mcar_perm", np.array([1, 0]), 2),
+    ]
+)
+def test__build_label(
+    request,
+    dataframe_fixture,
+    permutation_fixture,
+    features_idx,
+    target_idx
+):
+    dataframe = request.getfixturevalue(dataframe_fixture)
+    m_perm = request.getfixturevalue(permutation_fixture)
+    mcar_test_pklm = PKLMTest()
+    label = mcar_test_pklm._build_label(dataframe, m_perm, features_idx, target_idx)
+    assert not np.any(np.isnan(label))
+    assert len(label.shape) == 1
+    assert np.isin(label, [0, 1]).all()
+
+
+@pytest.mark.parametrize(
+        "oob_fixture, label",
+        [
+            ("oob_probabilities", np.array([1, 1, 1, 1])),
+            ("oob_probabilities", np.array([0, 0, 0, 0])),
+        ]
+)
+def test__U_hat_unique_label(request, oob_fixture, label):
+    oob_prob = request.getfixturevalue(oob_fixture)
+    mcar_test_pklm = PKLMTest()
+    mcar_test_pklm._U_hat(oob_prob, label)
+
+
+@pytest.mark.parametrize(
+        "oob_fixture, label, expected",
+        [
+            ("oob_probabilities", np.array([1, 0, 0, 0]), 2/3*(np.log(1 - 1e-9) - np.log(1e-9))),
+        ]
+)
+def test__U_hat_computation(request, oob_fixture, label, expected):
+    oob_prob = request.getfixturevalue(oob_fixture)
+    mcar_test_pklm = PKLMTest()
+    u_hat = mcar_test_pklm._U_hat(oob_prob, label)
+    assert round(u_hat, 2) == round(expected, 2)
+
+@pytest.mark.parametrize(
```

Comment:
just formatting: it could be good practice to config black formatter or ruff (locally, pre-commit, ....). What do you think ?